src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -172,30 +195,41 @@ public partial class DistributedApplicationExecutionContext
     {
         public DistributedApplicationExecutionContext(DistributedApplicationExecutionContextOptions options) { }
 
+        public DistributedApplicationExecutionContext(DistributedApplicationOperation operation, string publisherName) { }
+
         public DistributedApplicationExecutionContext(DistributedApplicationOperation operation) { }
 
+        public bool IsInspectMode { get { throw null; } }
```

Comment:
Are we concerned about all the checks for either `IsRunMode` or `IsPublishMode` and then `else` and now the `else` being taken for a new mode?

cc @mitchdenny

Reply:
Yes 😄

Reply:
Time for a switch expression:

```cs
builder.ExecutionContext.Operation switch
{
    DistributedApplicationOperation.Run => { ..run mode... },
    DistributedApplicationOperation.Publish => { ..pubish mode... },
    DistributedApplicationOperation.Inspect => { ..inspect mode... },
};
```

Oh wait, is that not a thing?

Reply:
Yes, it is something that I'm concerned about. When I added it I did a quick pass on our code base to see if I could spot anything that would be problematic. There were a few instances where tests failed etc. But we need to do some extensive deployment testing before we release to spot any more issues.